### ISSUE TITLE
Az upgrade lists different version than it actually installs

### ISSUE THREAD

**Issue by ajdvoynos** [Azure/azure-cli#25695]
## Describe the bug

**Command Name**
`az upgrade`

**Errors:**
```
Your current Azure CLI version is 2.25.0. Latest version available is 2.29.0
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Have installed az version 2.25 (I have not tested it with other versions)
- `az upgrade`

## Expected Behavior

```
Your current Azure CLI version is 2.25.0. Latest version available is 2.45.0
```

## Environment Summary (This was collected after running az upgrade, note that the version was correctly upgraded to 2.45 but the command said 2.29 would be installed)
```
Windows-10-10.0.22621-SP0
Python 3.10.8
Installer: MSI

azure-cli 2.45.0

Extensions:
communication 1.3.0

Dependencies:
msal 1.20.0
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
@bebound for awareness

**Comment by bebound** [comment]
Could you please provide the output of `az upgrade --debug` when you get `Your current Azure CLI version is 2.25.0. Latest version available is 2.29.0`

**Comment by ajdvoynos** [comment]
I cannot, as I've already upgraded to version 2.45. I only have the history output from the commands that I ran before/after running az upgrade, I've pasted them below if that's of any help.

```
❯ az --version
azure-cli                         2.25.0 *

core                              2.25.0 *
telemetry                          1.0.6

Extensions:
communication                      1.3.0

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\Andrei\.azure\cliextensions'

Python (Windows) 3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 13:22:56) [MSC v.1928 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


You have 2 updates available. Consider updating your CLI installation with 'az upgrade'

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy
```

```
❯ az upgrade
This command is in preview and under development. Reference and support levels: https://aka.ms/CLI_refstatus
Your current Azure CLI version is 2.25.0. Latest version available is 2.29.0.
Please check the release notes first: https://docs.microsoft.com/cli/azure/release-notes-azure-cli
Do you want to continue? (Y/n): Y
```

```
❯ az --version
azure-cli                         2.45.0

core                              2.45.0
telemetry                          1.0.8

Extensions:
communication                      1.3.0

Dependencies:
msal                              1.20.0
azure-mgmt-resource             21.1.0b1

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\Andrei\.azure\cliextensions'

Python (Windows) 3.10.8 (tags/v3.10.8:aaaf517, Oct 11 2022, 16:37:59) [MSC v.1933 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by bebound** [comment]
I figure it out.
In 2.25, CLI use this file to determine the latest version:
https://github.com/Azure/azure-cli/blob/3d14ca21e648f4fc0124132a43156a85a50251cd/src/azure-cli-core/azure/cli/core/util.py#L274
But the master branch has rename to main after 2.29.
